src/Controller.js

```diff
@@ -0,0 +1,125 @@
+(function exportController() {
+    class Controller {
+        constructor(ship) {
+            this.ship = ship;
+            this.initialiseSea();
+            this.waterState = false; // here false represents water0.png
+            
+            document.querySelector('#sailbutton').addEventListener('click', () => {
+                this.setSail();
+            })
+
+            this.renderHeadsUp();
+        }
+
+        initialiseSea() {
+            window.setInterval(this.changeWater, 1000)
+        }
+
+        changeWater() {
+            const viewport = document.querySelector('#viewport');
+
+            if(!this.waterState) {
+                viewport.style.backgroundImage = "url(../images/water1.png)";
+                this.waterState = true;
+            } else {
+                viewport.style.backgroundImage = "url(../images/water0.png)";
+                this.waterState = false;
+            }
+        }
+
+        renderPorts(ports) {
+            const portsElement = document.querySelector('#ports');
+            portsElement.style.width = '0px';
+
+            ports.forEach((port, index) => {
+                const newPortElement = document.createElement('div');
+                newPortElement.className = 'port';
+                newPortElement.dataset.portName = port.portName;
+                newPortElement.dataset.portIndex = index;
+
+                portsElement.appendChild(newPortElement); 
+                
+                const portsElementWidth = parseInt(portsElement.style.width, 10);
+                portsElement.style.width = `${portsElementWidth + 256}px`;
+            });
+        }
+
+        renderNewPort(port, index) {
+            const portsElement = document.querySelector('#ports');
+            const newPortElement = document.createElement('div');
+            index = this.ship.itinerary.ports.length-1;
+
+            newPortElement.className = 'port';
+            newPortElement.dataset.portName = port.portName;
+            newPortElement.dataset.portIndex = index;
+
+            portsElement.appendChild(newPortElement); 
+            
+            const portsElementWidth = parseInt(portsElement.style.width, 10);
+            portsElement.style.width = `${portsElementWidth + 256}px`;
+            this.renderHeadsUp();
+        }
+
+        renderShip() {
+            const shipPortIndex = this.ship.itinerary.ports.indexOf(this.ship.currentPort);
+            const portElement = document.querySelector(`[data-port-index='${shipPortIndex}']`);
```

Comment:
Dont need to assign this to a variable as its only used once:
```suggestion
            const portElement = document.querySelector(`[data-port-index='${this.ship.itinerary.ports.indexOf(this.ship.currentPort)}']`);
```